packages/compass-components/src/components/databases-navigation-tree.tsx

```diff
@@ -0,0 +1,712 @@
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  marginLeft: spacing[2],
+});
+
+const CollectionItem: React.FunctionComponent<
+  Omit<Collection, '_id'> & {
+    id: string;
+    posInSet: number;
+    setSize: number;
+    onNamespaceAction(namespace: string, action: Actions): void;
+    isActive: boolean;
+    isReadOnly: boolean;
+  }
+> = ({
+  id,
+  name,
+  type,
+  posInSet,
+  setSize,
+  onNamespaceAction,
+  isActive,
+  isReadOnly,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [focusProps, focusState] = useFocusState();
+  const [hoverProps, isHovered] = useHoverState();
+
+  const onDefaultAction = useCallback(
+    (evt) => {
+      onNamespaceAction(
+        evt.currentTarget.dataset.id as string,
+        'select-collection'
+      );
+    },
+    [onNamespaceAction]
+  );
+
+  const treeItemProps = useTreeItem({
+    level: 2,
+    setSize,
+    posInSet,
+    onDefaultAction,
+  });
+
+  const glyph = useMemo(() => {
+    return type === 'timeseries'
+      ? 'TimeSeries'
+      : type === 'view'
+      ? 'Visibility'
+      : 'Folder';
+  }, [type]);
+
+  const onMenuItemClick = useCallback(
+    (evt: React.MouseEvent<HTMLButtonElement>, action: Actions) => {
+      evt.stopPropagation();
+      setIsMenuOpen(false);
+      onNamespaceAction(id, action);
+    },
+    [id, setIsMenuOpen, onNamespaceAction]
+  );
+
+  const actions = useMemo(() => {
+    const actions: { action: Actions; label: string; icon?: string }[] = [
+      {
+        action: 'open-in-new-tab',
+        label: 'Open in New Tab',
+        icon: 'OpenNewTab',
+      },
+    ];
+
+    if (isReadOnly) {
+      return actions;
+    }
+
+    if (type === 'view') {
+      actions.push(
+        { action: 'drop-collection', label: 'Drop View' },
+        { action: 'duplicate-view', label: 'Duplicate View' },
+        { action: 'modify-view', label: 'Modify View' }
+      );
+    } else {
+      actions.push({ action: 'drop-collection', label: 'Drop Collection' });
+    }
+
+    return actions;
+  }, [type, isReadOnly]);
+
+  const isActionsVisible = isHovered || isActive || isMenuOpen;
+
+  return (
+    <li role="none" className={collectionListItem}>
+      <div
+        data-id={id}
+        data-testid={`collection-${id}`}
+        className={cx(
+          collectionItem,
+          focusRing,
+          focusState === FocusState.FocusVisible && focusRingVisible,
+          isHovered && !isActive && hoverBackground,
+          isActive && activeBackground
+        )}
+        {...focusProps}
+        {...hoverProps}
+        {...treeItemProps}
+      >
+        <span className={collectionItemIconContainer}>
+          <Icon size="small" glyph={glyph}></Icon>
+        </span>
+        <span className={collectionItemName}>{name}</span>
+        {isActionsVisible && (
+          <div className={actionsContainer} data-testid="collection-actions">
+            {actions.length > 1 ? (
```

Comment:
This is changed from previous behavior where we would just show some items as disabled, but my thinking here is that if it's just one item left in the context menu, why not just show it right away? I do need to check this with Claudia still

Reply:
Checked with Claudia, keeping this behavior